crates/ide_assists/src/handlers/inline_type_alias.rs

```diff
@@ -0,0 +1,714 @@
+        }
+    }
+
+    let instance_generics = generic_args_to_other_generics(instance_generic_args_list);
+    let alias_generics = generic_param_list_to_other_generics(&alias_generics);
+
+    if instance_generics.len() > alias_generics.len() {
+        cov_mark::hit!(too_many_generic_args);
+        return None;
+    }
+
+    // Any declaration generics that don't have a default value must have one
+    // provided by the instance.
+    for (i, declaration_generic) in alias_generics.iter().enumerate() {
+        let key = declaration_generic.replacement_key();
+
+        if let Some(instance_generic) = instance_generics.get(i) {
+            other_mappings.insert(key, instance_generic.replacement_value()?);
+        } else if let Some(value) = declaration_generic.replacement_value() {
+            other_mappings.insert(key, value);
+        } else {
+            cov_mark::hit!(missing_replacement_param);
+            return None;
+        }
+    }
+
+    let updated_concrete_type = concrete_type.clone_for_update();
+    let mut replacements = Vec::new();
+    let mut removals = Vec::new();
+
+    for syntax in updated_concrete_type.syntax().descendants() {
+        let syntax_string = syntax.to_string();
+        let syntax_str = syntax_string.as_str();
+
+        if syntax.kind() == SyntaxKind::LIFETIME {
+            let new = lifetime_mappings.get(syntax_str).expect("lifetime is mapped");
+            if new.text() == "'_" {
+                removals.push(NodeOrToken::Node(syntax.clone()));
+
+                if let Some(ws) = syntax.next_sibling_or_token() {
+                    removals.push(ws.clone());
+                }
+
+                continue;
+            }
+
+            replacements.push((syntax.clone(), new.syntax().clone_for_update()));
+        } else if let Some(replacement_syntax) = other_mappings.get(syntax_str) {
+            let new_string = replacement_syntax.to_string();
+            let new = if new_string == "_" {
+                make::wildcard_pat().syntax().clone_for_update()
+            } else {
+                replacement_syntax.clone_for_update()
+            };
+
+            replacements.push((syntax.clone(), new));
+        }
+    }
+
+    for (old, new) in replacements {
+        ted::replace(old, new);
+    }
+
+    for syntax in removals {
+        ted::remove(syntax);
+    }
+
+    Some(updated_concrete_type.to_string())
+}
+
+fn get_type_alias(ctx: &AssistContext, path: &ast::PathType) -> Option<ast::TypeAlias> {
+    let resolved_path = ctx.sema.resolve_path(&path.path()?)?;
+
+    // We need the generics in the correct order to be able to map any provided
+    // instance generics to declaration generics. The `hir::TypeAlias` doesn't
+    // keep the order, so we must get the `ast::TypeAlias` from the hir
+    // definition.
+    if let PathResolution::Def(hir::ModuleDef::TypeAlias(ta)) = resolved_path {
+        ast::TypeAlias::cast(ctx.sema.source(ta)?.syntax().value.clone())
```

Comment:
```suggestion
        ctx.sema.source(ta)?.value
```
shouldn't this just work?